Libraries/Opc.Ua.Server/Diagnostics/CustomNodeManager.cs

```diff
@@ -4206,38 +4206,12 @@ protected virtual void OnDeleteMonitoredItemsComplete(ServerSystemContext contex
             IList<IMonitoredItem> transferredItems = new List<IMonitoredItem>();
             lock (Lock)
             {
-                for (int ii = 0; ii < monitoredItems.Count; ii++)
-                {
-                    // skip items that have already been processed.
-                    if (processedItems[ii] || monitoredItems[ii] == null)
-                    {
-                        continue;
-                    }
-
-                    // check handle.
-                    NodeHandle handle = IsHandleInNamespace(monitoredItems[ii].ManagerHandle);
-                    if (handle == null)
-                    {
-                        continue;
-                    }
-
-                    // owned by this node manager.
-                    processedItems[ii] = true;
-                    var monitoredItem = monitoredItems[ii];
-                    transferredItems.Add(monitoredItem);
-
-                    if (sendInitialValues && !monitoredItem.IsReadyToPublish)
-                    {
-                        if (monitoredItem is IDataChangeMonitoredItem2 dataChangeMonitoredItem)
-                        {
-                            errors[ii] = ReadInitialValue(systemContext, handle, dataChangeMonitoredItem);
-                        }
-                    }
-                    else
-                    {
-                        errors[ii] = StatusCodes.Good;
-                    }
-                }
+                transferredItems = TransferMonitoredItems(
+                    systemContext,
+                    sendInitialValues,
```

Comment:
the signature of the internal Transfer function only differs by context variable, it should be better distinguishable for a caller or maybe private ?